pkgs/applications/version-management/mercurial/4.9.nix

```diff
@@ -1,21 +1,19 @@
 { lib, stdenv, fetchurl, python2Packages, makeWrapper
 , guiSupport ? false, tk ? null
 , ApplicationServices
-, mercurialSrc ? fetchurl rec {
-    meta.name = "mercurial-${meta.version}";
-    meta.version = "4.9.1";
-    url = "https://mercurial-scm.org/release/${meta.name}.tar.gz";
-    sha256 = "0iybbkd9add066729zg01kwz5hhc1s6lhp9rrnsmzq6ihyxj3p8v";
-  }
 }:
 
 let
   inherit (python2Packages) docutils hg-git dulwich python;
 
-in python2Packages.buildPythonApplication {
+in python2Packages.buildPythonApplication rec {
+  pname = "mercurial";
+  version = "4.9.1";
```

Comment:
According to official page, Mercurial 5.2 accepts Python 3.

Just a reminder.

Reply:
https://github.com/NixOS/nixpkgs/pull/76903